diff --git memcache_admin/memcache_admin.module memcache_admin/memcache_admin.module
index 9bba570..b796396 100644
--- memcache_admin/memcache_admin.module
+++ memcache_admin/memcache_admin.module
@@ -408,6 +408,8 @@ function theme_memcache_admin_stats_table($bin, $servers, $stats) {
 }
 
 function memcache_admin_stats_types($bin) {
+  module_load_include('inc', 'memcache', 'dmemcache');
+
   if ($mc = dmemcache_object($bin)) {
     if ($mc instanceof Memcache) {
       // TODO: Determine which versions of the PECL memcache extension have
