diff --git a/includes/content.inc b/includes/content.inc
index e155dbe..c2e5e49 100644
--- a/includes/content.inc
+++ b/includes/content.inc
@@ -322,14 +322,16 @@ function content_wysiwyg_fields_nodeapi_alter($node, $op, $a3 = NULL, $a4 = NULL
         }
 
         // Re-assign deltas to match CCKs changes.
-        foreach ($info['fields'] as $field_name => $field) {
-          foreach (array_keys($field['deltas']) as $delta) {
-            foreach ($node->{$field_name} as $field_delta => $item) {
-              if ($item['wysiwyg_fields']['wysiwyg_fields_delta'] == $delta && $delta != $field_delta) {
-                foreach ($info['fields'][$field_name]['deltas'][$delta] as &$reference) {
-                  $reference = $field_delta;
+        if (is_array($info['fields'])) {
+          foreach ($info['fields'] as $field_name => $field) {
+            foreach (array_keys($field['deltas']) as $delta) {
+              foreach ($node->{$field_name} as $field_delta => $item) {
+                if ($item['wysiwyg_fields']['wysiwyg_fields_delta'] == $delta && $delta != $field_delta) {
+                  foreach ($info['fields'][$field_name]['deltas'][$delta] as &$reference) {
+                    $reference = $field_delta;
+                  }
+                  break;
                 }
-                break;
               }
             }
           }
@@ -354,23 +356,25 @@ function content_wysiwyg_fields_nodeapi_alter($node, $op, $a3 = NULL, $a4 = NULL
         }
 
         // Set new deltas.
-        foreach (array_keys($info['tokens']) as $token) {
-          $info['tokens'][$token]['wf_deltas'] = array_flip($info['tokens'][$token]['wf_deltas']);
-          foreach (array_keys($info['tokens'][$token]['wf_deltas']) as $delta) {
-            $info['tokens'][$token]['wf_deltas'][$delta] = $delta;
+        if (is_array($info['tokens'])) {
+          foreach (array_keys($info['tokens']) as $token) {
+            $info['tokens'][$token]['wf_deltas'] = array_flip($info['tokens'][$token]['wf_deltas']);
+            foreach (array_keys($info['tokens'][$token]['wf_deltas']) as $delta) {
+              $info['tokens'][$token]['wf_deltas'][$delta] = $delta;
+            }
+            ksort($info['tokens'][$token]['wf_deltas']);
           }
-          ksort($info['tokens'][$token]['wf_deltas']);
-        }
 
-        // Replace tokens.
-        foreach ($info['tokens'] as $token => $attributes) {
-          foreach ($textfields as &$textfield) {
-            $attributes['wf_deltas'] = is_array($attributes['wf_deltas']) ? implode(',', array_keys($attributes['wf_deltas'])) : $attributes['wf_deltas'];
-            $new_token = array();
-            foreach ($attributes as $key => $value) {
-              $new_token[] = "{$key}=\"{$value}\"";
+          // Replace tokens.
+          foreach ($info['tokens'] as $token => $attributes) {
+            foreach ($textfields as &$textfield) {
+              $attributes['wf_deltas'] = is_array($attributes['wf_deltas']) ? implode(',', array_keys($attributes['wf_deltas'])) : $attributes['wf_deltas'];
+              $new_token = array();
+              foreach ($attributes as $key => $value) {
+                $new_token[] = "{$key}=\"{$value}\"";
+              }
+              $textfield['value'] = str_replace($token, '[wysiwyg_field ' . implode(' ', $new_token) . ']', $textfield['value']);
             }
-            $textfield['value'] = str_replace($token, '[wysiwyg_field ' . implode(' ', $new_token) . ']', $textfield['value']);
           }
         }
       }
