diff --git a/includes/wysiwyg.inc b/includes/wysiwyg.inc
index 2523183..db28b65 100644
--- a/includes/wysiwyg.inc
+++ b/includes/wysiwyg.inc
@@ -118,6 +118,6 @@ function wysiwyg_fields_wysiwyg_plugin_callback() {
  */
 function wysiwyg_fields_wysiwyg_editor_settings_alter(&$settings, $context) {
   if ($context['profile']->editor == 'tinymce'/* && $context['profile']->settings['buttons']['drupal']['MY_PLUGIN']*/) {
-    $settings['extended_valid_elements'] = (!empty($settings['extended_valid_elements']) ? $settings['extended_valid_elements'] . ', ' : '') . 'wysiwyg_field[*]';
+    $settings['extended_valid_elements'] = (!empty($settings['extended_valid_elements']) ? $settings['extended_valid_elements'] . ',' : '') . 'wysiwyg_field[*]';
   }
 }
