Index: purl.module
===================================================================
--- purl.module	(revision 4924)
+++ purl.module	(revision 4927)
@@ -683,6 +683,8 @@
     '#tree' => TRUE,
     '#element_validate' => array('purl_form_validate'),
   );
+  
+  if(!$processor) return $form;
 
   $processors = _purl_options();
 
@@ -878,7 +880,7 @@
   if (!isset($processors[$method])) {
     ctools_include('plugins');
     $class = ctools_plugin_load_class('purl', 'processor', $method, 'handler');
-    $processors[$method] = new $class();
+    if($class) $processors[$method] = new $class();
   }
   return isset($processors[$method]) ? $processors[$method] : NULL;
 }
